radixdlt-core/docs/api-info.md

```diff
@@ -0,0 +1,184 @@
+ | /system/info | GET | Get system information - public key, agent, supported protocols | No sensitive information |
+ | /system/checkpoints | GET | Get genesis txn and proof | No sensitive information |
+ | /system/proof | GET | Get current proof | No sensitive information |
+ | /system/epochproof | GET | Get epoch proof | No sensitive information |
+ | /system/peers | GET | Get information about peer nodes | No sensitive information |
+ | /universe.json | GET | Get Radix Universe | No sensitive information |
+
+### JSON-RPC APIs
+
+| Method | Description |
+| --- | --- |
+| radix.nativeToken | Information about native token |
+| radix.tokenInfo | Information about specified token |
+| radix.tokenBalances | Token balances at specified account |
+| radix.stakePositions | Stakes made from specified account |
+| radix.unstakePositions | Pending unstakes for specified account|
+| radix.transactionHistory | Transaction history for specified account |
+| radix.lookupTransaction | Lookup specified transaction |
+| radix.statusOfTransaction | Status of specified transaction |
+| radix.buildTransaction | Assemble transaction given specified actions and optional message |
+| radix.finalizeTransaction | Send signed transaction and receive transaction ID|
+| radix.submitTransaction | Submit signed transaction for processing by network|
+| radix.validators | Get list of validators |
+| radix.lookupValidator | Get information about specified validator |
+| radix.networkId | Get network ID |
+| radix.networkTransactionThroughput | Get number of transactions per second |
+| radix.networkTransactionDemand | Get average number of transactions waiting for processing in mempool |
+
+### Configuration Options
+
+Following configuration options control which APIs are enabled at the node:
+
+| Configuration Option | Description |
+| --- | --- |
+| client_api.enable | Enables JSON-RPC APIs|
+| universe_api.enable | Enables `/universe.json` API |
+
+</details>
+
+## New Organization of APIs
+
+ The general approach to the API organization is based on the following considerations:
+- Most methods are spread across five endpoints:
+  - `/archive` (formerly `/rpc`) - read-only methods, requiring an archive node
+  - `/construction` - methods which support building and submitting transactions, which can be enabled by either archive or full nodes
+  - `/account` supports methods to fetch your associated account info, and a one-step method to build, sign, and submit a transaction
+  - `/validation` - read-only methods which provide same information as available today via all `/node/*` endpoints
+  - `/system` - read-only methods which provide same information as available today via all `/system/*` endpoints
+    
+- The `/system`, `/account` and `/validator` endpoints are expected to be protected by firewall and/or require authentication/etc. 
+  (similar requirements/setup as we have today)
+
+### REST APIs
```

Comment:
The old REST APIs are not RESTful, but the ones that have been kept are REST-compliant imo.

My suggestion for this doc is probably to remove the specs about old APIs. It would be easier for new people to digest the new specs than to trace the API changes (our internal people doing migration work might have a different say though).

Reply:
Fixed